### ISSUE TITLE
[Bug] Fenix shows custom tabs when opened, and does not exit them cleanly

### ISSUE THREAD

**Issue by Manishearth** [mozilla-mobile/fenix#8553]
Probably the same underlying bug as #8554

## Steps to reproduce
 
(Fenix must be the default browser)

 1. Open any app that has external links (Twitter, Slack, Github, GMail, etc)
 2. Click a link in the app to bring up a Fenix custom tab
 3. Exit the app via the home button. 
 4. Open Fenix from the home screen. The custom tab will be visible
 5. Exit the custom tab by tapping the "x" or hitting back

### Expected behavior

I'd expect that (4) would _not_ show the custom tab (the custom tab belongs to the app, not the browser, I think this is how other browsers do it too and it's definitely how Fennec used to work).

Even if (4) didn't work, tapping "back" or exiting the custom tab should take you to the browser home.

### Actual behavior

After (4) the custom tab is still visible.

After (5), the entire browser exits. This means that if I was scrolling through a custom tab somewhere earlier, and open Fenix to browse, I have to exit the tab and then open Fenix _again_, which is very weird.

### Device information

* Android device: Pixel 3
* Fenix version: 3.2.1 (Build #20250350) 28.0.3, f0dc2cadd, Saturday 1/25 @ 3:49 AM

**Comment by Manishearth** [comment]
Just tested this on Firefox Preview Nightly and I cannot reproduce this exact bug (@darkwing , are you seeing this as well on your master build?)

However, a smaller form of this bug still exists, where if I:

 - Open Twitter (etc)
 - Open an external link in Twitter into a custom tab
 - Switch to a different app
 - Reopen twitter from the home screen
 - Close the custom tab (pressing back or "x")

it exits the app instead of bringing me back to where I was before I opened the custom tab. This could potentially be an Android OS issue though

**Comment by Manishearth** [comment]
Given that I can't repro on nightly, closing for now. Filed #8557 with the other bug I noticed

**Comment by madb1lly** [comment]
Hi all,

I think I have a subset of the experience in the issue reported. When I open a link in a custom tab and then ask to open in nightly, then I use task switcher, I _sometimes_ still have the custom tab visible in the task switcher, but not always. I will try and test steps 3, 4 and 5 above next time I experience the custom tab still visible in the task switcher.

Note that I found this was _always_ the behaviour in Fennec, and is one of the reasons (there are others) why I found the tab queue so helpful in Fennec, which is still not available in Fenix.

Cheers 🙂
